### PR TITLE
Feature: Extract responsibility of mapping request to menu model

### PR DESCRIPTION
**Pull request by lbajsarowicz**
With the current approach, when you want to add an extra attribute to Menu, you cannot achieve that without inheritance of `Save` controller, you cannot achieve that.

I have introduced the `Hydrator` service, that enables you to map Request elements to Menu model.
This way you can apply `after` plugin that maps extra fields to the menu.

This is the 1st step of upcoming changes, as I'd love to introduce such hydrator also for Nodes.

### CODE REVIEW

**Comment by snowremi** [comment]
Internal ticket DEV-69365

**Comment by TM18** [comment]
@lbajsarowicz was this ticket closed on purpose?

**Comment by lbajsarowicz** [comment]
A month passed by without any feedback.
I'm going to fork and develop the MegaMenu on my own.

**Comment by lbajsarowicz** [comment]
giphy

Don't make it hang too long.

File: Controller/Adminhtml/Menu/Index.php [hunk 1]
```diff
@@ -1,29 +1,19 @@
 <?php
+declare(strict_types=1);
+
 namespace Snowdog\Menu\Controller\Adminhtml\Menu;
 
 use Magento\Backend\App\Action;
 use Magento\Backend\App\Action\Context;
+use Magento\Framework\Controller\ResultFactory;
 use Magento\Framework\View\Result\PageFactory;
```
Comment (TM18) [inline Controller/Adminhtml/Menu/Index.php:9]: `Context` and `PageFactory` are not used anymore so I think they can be safely removed from here.

**Review by TM18 — CHANGES_REQUESTED** [review]

File: Controller/Adminhtml/Menu/Index.php (hunk 1 above)
Comment (lbajsarowicz) [reply Controller/Adminhtml/Menu/Index.php:9]: Yes, you are right. I'm going to adjust that later afternoon.

**Review by TM18 — APPROVED** [review]

**Pull request merged**